### DISCUSSION TITLE
How to validate a Supabase JWT token in my nodejs backend

### DISCUSSION THREAD

**Discussion by AdrianKrebs** [supabase/supabase#2873] [Questions] [answered]
I'm using supabase authentication in my frontend which communicates with a separate backend.
I would like to validate the JWT token of the authenticated Supabase user in my backend. How can I do that?

According to this issue https://github.com/supabase/supabase/issues/147, it should be possible to do 
`const response = await supabase.auth.user({ jwt: "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.xxx" })`
But that doesn't seem to be part of the library yet. Any suggestions?

**Comment by dshukertjr** [comment, marked as answer]
Hi @AdrianKrebs!

You can accomplish it like this:
```typescript
const { data, error } = supabaseAdmin.auth.api.getUser('ACCESS_TOKEN')
```

You can see more examples in the Supabase Stripe example [here](https://github.com/vercel/nextjs-subscription-payments/blob/main/utils/supabase-admin.js)

**Reply by AdrianKrebs** [reply]
Thanks! Any experience in how fast this call is? If I need to validate the token for every backend request, this check should be performant.

**Reply by dshukertjr** [reply]
@AdrianKrebs 
Everything in Supabase is generally super performant, so I don't really have any bench marks, but I don't think it should concern you!
